src/plugins/sockets/inGameEvents.js

```diff
@@ -125,57 +126,47 @@ export async function handleInGameEvents(evData) {
       break;
     case SocketEvent.REMATCH:
       gameStore.setRematch({ pNum: evData.pNum, rematch: evData.game[`p${evData.pNum}Rematch`] });
+      return;
+    case SocketEvent.NEW_GAME_FOR_REMATCH: {
+      // ignore if not currently in/spectating relevant game
       if (
-        evData.pNum === gameStore.myPNum &&
-        currentRoute.name !== ROUTE_NAME_SPECTATE &&
-        evData.game[`p${evData.pNum}Rematch`]
+        Number(urlGameId) !== evData.oldGameId || 
+        ![ROUTE_NAME_GAME, ROUTE_NAME_SPECTATE].includes(currentRoute.name)
       ) {
-        router.push({
-          name: ROUTE_NAME_REMATCH,
-          params: {
-            gameId: gameStore.id,
-          },
-        });
+        return;
       }
-      return;
-    case SocketEvent.NEW_GAME_FOR_REMATCH: {
-      gameStore.setRematchGameId(evData.gameId);
+      gameStore.p0Rematch = true;
+      gameStore.p1Rematch = true;
+      gameStore.rematchGameId = evData.gameId;
+      // stop early if spectating and haven't yet chosen to continue spectating
+      if (gameStore.isSpectating && !gameStore.iWantToContinueSpectating) {
+        return;
+      }
+
+      // wait for card flip animations
+      await sleep(500);
```

Comment:
This is only used here, I'd argue you don't need a utility and instead should inline the promise check. It would be better to file a follow up to use the vue transition component and the hooks it provides instead of a manual wait though, see https://stackblitz.com/edit/vue-3-card-flip-animation?file=src%2FApp.vue,src%2Fcomponents%2FCard.vue.

Reply:
Per our discord convo, we'll keep this for the moment to avoid indenting these. We should roll out `sleep()` across the existing setTimeout use cases then rework transitions across the board to properly utilize `Transition` components